### PR TITLE
 Add module for Cisco IOS configuration manipulation 

### PR DESCRIPTION
**Pull request by mirceaulinic**
### What does this PR do?

Adds a new module for Cisco IOS style configuration manipulation, and use the functionality into the NAPALM module to have the features directly applied on the configuration(s) of the managed device.

### CODE REVIEW

File: salt/modules/iosconfig.py [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+This module provides a collection of helper functions for Cisco IOS style
+configuration manipulation. This module does not have external dependencies
+and can be used from any Proxy or regular Minion.
+'''
+# Import Python Libs
+from __future__ import absolute_import, unicode_literals, print_function
+
+# Import python stdlib
+import difflib
+from collections import OrderedDict
```
Comment (rallytime) [inline salt/modules/iosconfig.py]: This should use the `OrderedDicrt` from `salt.utils.odict`.

File: tests/unit/modules/test_iosconfig.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
+# -*- coding: utf-8 -*-
+'''
+Test the iosconfig Execution module.
+'''
+from __future__ import absolute_import, print_function, unicode_literals
+
+# Import Python libs
+import textwrap
+from collections import OrderedDict
```
Comment (rallytime) [inline tests/unit/modules/test_iosconfig.py]: Same comment here. :)

**Review by rallytime — CHANGES_REQUESTED** [review]
I have 2 small requests. Thank you for writing tests for this! 🎉 

There's also a pylint warning.

**Comment by mirceaulinic** [comment]
Wasn't aware of `salt.utils.odict` (or probably I forgot) - should be fixed via https://github.com/saltstack/salt/pull/48859/commits/062f161a7d61bd21eaded5ab3688deebb784f439 @rallytime.

**Comment by rallytime** [comment]
@mirceaulinic Can you fix the lint error too? 

https://jenkinsci.saltstack.com/job/pr-lint/job/PR-48859/2/warnings52Result/

**Comment by mirceaulinic** [comment]
Alright, seems like https://github.com/saltstack/salt/pull/48859/commits/ca254fbdfe224bbeb9ef5fa639a42e6663fb2241 fixed the lint.

**Review by rallytime — APPROVED** [review]

**Pull request merged**